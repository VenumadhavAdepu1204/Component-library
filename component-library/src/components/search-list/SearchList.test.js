import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'

import SearchList from './index'
import CardListJson from '../stubs/cardList.json'

jest.mock('../../../IB/components/icon', () => 'Icon')

jest.mock(
  '@anz/icon/dist/line/arrows-and-symbols/information',
  () => 'InfoIcon'
)

describe('SearchList', () => {
  afterEach(cleanup)

  test('renders properly', () => {
    const { container } = render(
      <SearchList list={CardListJson.cards} renderList={jest.fn()} />
    )
    expect(container).toMatchSnapshot()
  })

  test('on enter of filter text', () => {
    const { getByPlaceholderText } = render(
      <SearchList
        cardlist={CardListJson.cards}
        filterPlaceholder='search'
        handleFilter={jest.fn()}
        renderList={jest.fn()}
      />
    )
    fireEvent.change(getByPlaceholderText(/search/), { target: { value: 's' } })
  })

  test('on click of cross icon', () => {
    const { getByPlaceholderText, getByTestId } = render(
      <SearchList
        cardlist={CardListJson.cards}
        filterPlaceholder='search'
        handleFilter={jest.fn()}
        renderList={jest.fn()}
      />
    )
    fireEvent.change(getByPlaceholderText(/search/), { target: { value: 's' } })
    fireEvent.click(getByTestId(/clear-icon/))
  })
})
