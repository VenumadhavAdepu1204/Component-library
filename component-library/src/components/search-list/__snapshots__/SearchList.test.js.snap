// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchList renders properly 1`] = `
<div>
  <div
    class="sc-gqjmRU iiGbTV"
  >
    <div
      class="sc-VigVT gEhZuK"
    >
      <a
        class="sc-jTzLTM cLJNlm"
      >
        <icon
          iconname="Search"
        />
      </a>
      <div
        class="sc-dnqmqq ihJJte"
      >
        <div
          class="sc-ifAKCX bDiBLt"
        >
          <input
            aria-label=""
            aria-required="false"
            autocomplete="off"
            data-required="false"
            data-test-id="search-input"
            data-testid="search-input"
            id="search-input"
            name="search-input"
            placeholder=""
            tabindex="0"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
